solana/pyth2wormhole/program/src/config.rs

```diff
@@ -10,6 +10,8 @@ pub struct Pyth2WormholeConfig {
     pub wh_prog: Pubkey,
     /// Authority owning Pyth price data
     pub pyth_owner: Pubkey,
+    /// How many product/price pairs can be sent and attested at once
+    pub max_batch_size: u16,
```

Comment:
is there a check at config loading time that this number is < P2W_MAX_BATCH_SIZE ?

Reply:
Currently, it effectively *is* P2W_MAX_BATCH_SIZE and is only stored as courtesy toward attesters - the number of symbols expected by the business logic is maintained manually in `attest.rs` (Solana framework limitation that can be lifted in the future) and hard-coded in pyth2wormhole-client config-related options. The idea is that attesters read it right before deciding how to divide their symbols into batches. If we upgrade the contract with a new max_batch_size (especially changing to a smaller one), all attesters will be able to use the correct value immediately after the change is reflected on-chain in this struct.

Reply:
Got it, so we have to be careful to make sure to include config change transactions with upgrade to avoid downtime with this approach.

Reply:
Yes, batching is a special snowflake in this respect. I was not able to back the config value with logic that would respect it dynamically (Option is not enough for that, we would need Vec<>). Otherwise I think the right thing to do is have the program understand the preceding config and use the older one until the privileged set-config call is made. This is an actual hard requirement as the program must understand an older config enough to verify the authority attempting to migrate it.